### PR TITLE
[IE TESTS][CPU BF16] Added support for the eltwise operation with bf16 precision in the single layer tests.

### PR DESCRIPTION
**Pull request by maxnick**
The aim of this PR is to incorporate bfloat16 precision into the CPU single layer tests infrastructure. This was done using the eltwise layer as an example.

The main changes are as follows:
1. Support for bf16 input was added to MKLDNN plugin. In order to run an entire network with bf16 precision.
2. BF16 precision was added to the Eltwise CPU single layer tests, and Logical and Activation CPU single layer tests was created.
3. Added automatic skip for bf16 precision on platforms that do not support such operations.
4. Some data and blob utils extended with bf16 support.
5. Implicit BF16 optimizations are disabled for CPU single layer tests. To always run the test layer with the specified precision.
6. Conversion of input blobs was added to the LayerTestCommon reference calculation routine. Because ngraph evaluate routine does not support all possible precisions, and there is no need to obtain reference data of the specified type.
7. The makeInputLayer routine now fills the generated constants with 3 instead of zeroes, as it was before, to avoid removing some test layers during optimization. For example, when we add a null constant or multiplying by a constant filled with ones.
8. To enable constant folding with element types bf16 in ngraph, instances of the evaluate template method have been added to some ngraph ops, which increases the size of libngraph.so by 20KB. (With gcc 7.5.0 compiler).
9. Check for broadcast rule has been added to PowerToPowerIE transformation.

### CODE REVIEW

**Review by alexey-varyzgin — APPROVED** [review]

File: inference-engine/tests/functional/plugin/shared/src/single_layer_tests/activation.cpp [hunk 1]
```diff
@@ -104,7 +111,30 @@ InferenceEngine::Blob::Ptr ActivationLayerTest::GenerateInput(const InferenceEng
                                             32768);
 }
 
-ngraph::ParameterVector ActivationParamLayerTest::createActivationParams(ngraph::element::Type ngPrc, std::vector<size_t> inShape) {
+void ActivationLayerTest::setActivationExpectedPrc(std::string name, InferenceEngine::Precision netPrc) {
```
Comment (iefode) [inline inference-engine/tests/functional/plugin/shared/src/single_layer_tests/activation.cpp:114]: Looks custom case only for BF16. I don't like it

File: inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp [hunk 2]
```diff
@@ -97,7 +101,13 @@ void LayerTestsCommon::ConfigureNetwork() const {
 }
 
 void LayerTestsCommon::LoadNetwork() {
+    using namespace InferenceEngine;
+    if (configuration.find(PluginConfigParams::KEY_ENFORCE_BF16) == configuration.end()) {
```
Comment (iefode) [inline inference-engine/tests/ie_test_utils/functional_test_utils/layer_test_utils.cpp:105]: Again. Custom case for bf16. It works only for CPU

File: inference-engine/tests/functional/plugin/shared/include/single_layer_tests/activation.hpp [hunk 3]
```diff
@@ -88,6 +88,7 @@ class ActivationLayerTest : public testing::WithParamInterface<activationParams>
 
 protected:
     void SetUp() override;
+    void setActivationExpectedPrc(std::string name, InferenceEngine::Precision netPrc);
```
Comment (iefode) [inline inference-engine/tests/functional/plugin/shared/include/single_layer_tests/activation.hpp:91]: Looks like 'Private' method

File: inference-engine/tests/functional/plugin/shared/src/single_layer_tests/eltwise.cpp [hunk 4]
```diff
@@ -71,11 +71,32 @@ void EltwiseLayerTest::SetUp() {
...
 
-    auto secondaryInput = ngraph::builder::makeInputLayer(ngPrc, secondaryInputType, shape_input_secondary);
-    if (secondaryInputType == ngraph::helpers::InputLayerType::PARAMETER) {
-        input.push_back(std::dynamic_pointer_cast<ngraph::opset3::Parameter>(secondaryInput));
+    std::shared_ptr<ngraph::Node> secondaryInput;
+    if (eltwiseType == ngraph::helpers::EltwiseTypes::DIVIDE ||
+        eltwiseType == ngraph::helpers::EltwiseTypes::FLOOR_MOD ||
+        eltwiseType == ngraph::helpers::EltwiseTypes::MOD) {
+        std::vector<float> data(ngraph::shape_size(shape_input_secondary));
+        data = NGraphFunctions::Utils::generateVector<ngraph::element::Type_t::f32>(ngraph::shape_size(shape_input_secondary), 10, 2);
```
Comment (iefode) [inline inference-engine/tests/functional/plugin/shared/src/single_layer_tests/eltwise.cpp:79]: Can you use borders for vector in this function?

**Comment by maxnick** [comment]
The PR closed because a separate branch for bfloat16 development was set up.

**Pull request closed without merging**